models/knownorigin/knownorigin_ethereum_events.sql

```diff
@@ -0,0 +1,125 @@
+{{ config(
+    alias = 'events',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_time', 'unique_trade_id'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "knownorigin",
+                                \'["cat"]\') }}'
+    )
+}}
+
+with raw_ko_transactions as (
+    select  a.evt_block_time
+            , b.contract_address
+            , _tokenId as token_id
+            , _seller as seller 
+            , _buyer as buyer
+            , _price as raw_amount
+            , a.evt_block_number
+            , a.evt_index
+            , a.evt_tx_hash
+            , case when b.from = '0x0000000000000000000000000000000000000000' then 'Primary' else 'Secondary' end as sale_type
+            , 'Buy' as trade_category
+            , lower('0xc322cdd03f34b6d25633c2abbc8716a058c7fe9e') as project_contract_address
+    from {{ source('knownorigin_ethereum','TokenMarketplaceV2_evt_TokenPurchased') }} a
+    inner join {{ source('erc721_ethereum','evt_transfer') }} b on a.evt_tx_hash = b.evt_tx_hash and a._tokenId = b.tokenId
```

Comment:
add incremental filter on `evt_transfer` within join conditions, similar to prior PR we did, plz 🤝

Reply:
same idea throughout